### PR TITLE
Store login token

### PR DESCRIPTION
**Pull request by ryhuz**
storing login token on login context and localstorage
redirect to shelter home from login page if already logged in
re-naming URLs for `pet/add` to include shelterId
appending shelterId to add pet body

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
 		"react": "^17.0.2",
 		"react-dom": "^17.0.2",
 		"styled-components": "^5.3.0",
+		"uuid-validate": "^0.0.3",
```
Comment (stonefruit) [inline frontend/package.json:43]: Would prefer to use own function instead of relying on like unmaintained package.

**Comment by stonefruit** [comment]
Quite a lot of files changed so will probably have to test on weekend.

For now it seems cleaning of localstorage will only be done when logout button is pressed, and will not happen if api call returns 401.

File: frontend/package.json (hunk 1 above)
Comment (ryhuz) [reply frontend/package.json:43]:
```
function checkIfValidUUID(str) {
  // Regular expression to check if string is a valid UUID
  const regexExp = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;

  return regexExp.test(str);
}
```
can?

Comment (stonefruit) [reply frontend/package.json:43]: I'm not sure. This is the kind of thing that should have a unit test.

**Pull request merged**